iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/AffMatchingQualityTest.java

```diff
@@ -0,0 +1,279 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.io.Files;
+
+import eu.dnetlib.iis.importer.schemas.Organization;
+import eu.dnetlib.iis.importer.schemas.ProjectToOrganization;
+import eu.dnetlib.iis.metadataextraction.schemas.ExtractedDocumentMetadata;
+import eu.dnetlib.iis.referenceextraction.project.schemas.DocumentToProject;
+import eu.dnetlib.iis.wf.affmatching.model.MatchedOrganization;
+import eu.dnetlib.iis.wf.affmatching.model.SimpleAffMatchResult;
+import pl.edu.icm.sparkutils.test.SparkJob;
+import pl.edu.icm.sparkutils.test.SparkJobBuilder;
+import pl.edu.icm.sparkutils.test.SparkJobExecutor;
+
+/**
+ * Affiliation matching module test that measures quality of matching.<br/>
+ * Tests in this class completely ignores {@link MatchedOrganization#getMatchStrength()}<br/>
+ * <br/>
+ * Quality of matching is described by two factors:<br/>
+ * <ul>
+ * <li>Correct matches - percentage of expected matches that are also present in returned results</li>
+ * <li>False positives - percentage of returned matches that was matched, but incorrectly (are not present in expected matches)</li>
+ * <ul><br/>
+ * 
+ * @author madryk
+ */
+public class AffMatchingQualityTest {
+    
+    
+    private SparkJobExecutor executor = new SparkJobExecutor();
+    
+    private File workingDir;
+    
+    private String inputOrgDirPath;
+    
+    private String inputAffDirPath;
+    
+    private String inputDocProjDirPath;
+    
+    private float inputDocProjConfidenceThreshold = 0.8f;
+    
+    private String inputProjOrgDirPath;
+    
+    private String outputDirPath;
+    
+    
+    @Before
+    public void before() {
+        
+        workingDir = Files.createTempDir();
+        
+        inputOrgDirPath = workingDir + "/affiliation_matching/input/organizations";
+        inputAffDirPath = workingDir + "/affiliation_matching/input/affiliations";
+        inputDocProjDirPath = workingDir + "/affiliation_matching/input/doc_proj";
+        inputProjOrgDirPath = workingDir + "/affiliation_matching/input/proj_org";
+        outputDirPath = workingDir + "/affiliation_matching/output";
+        
+        
+    }
+    
+    
+    @After
+    public void after() throws IOException {
+        
+        FileUtils.deleteDirectory(workingDir);
+        
+    }
+    
+    
+    //------------------------ TESTS --------------------------
+    
+    @Test
+    public void affiliationMatchingJob_random_docs() throws IOException {
+        
+        
+        // given
+        
+        createInputDataFromJsonFiles(
+                of("src/test/resources/experimentalData/input/all_organizations.json"),
+                of("src/test/resources/experimentalData/input/set1/docs_with_aff_real_data.json"),
+                of(), of());
+        
+        
+        // execute
+        
+        executeJob();
+        
+        
+        // assert
+        
+        System.out.println("\nRANDOM DOCUMENTS");
```

Comment:
Do we assert anything here? I guess we only log it. Maybe we should introduce more appropriate section name: `log`? This will also indicate we do not check results at all in this test.